test/test_tools.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+set -e
+set -x
+
+srcdir=`dirname $0`/..
+
+on_error() {
+    rm -fr ./test_results_temp/
+}
+trap on_error ERR INT TERM
+
+mkdir -p ./test_results_temp/
```

Comment:
Tests should execute in a temporary directory, otherwise they'll clobber each other on multiple executions.